app/src/main/java/com/example/bikebuddy/Data/DbHelper.java

```diff
@@ -0,0 +1,294 @@
+            workout.setID(id);
+            workout.setDate(date);
+            workout.setTime(time);
+            workout.setListHR(heartRate);
+            workout.setListSpeed(speed);
+            workout.setTotalDuration(duration);
+            workout.setTotalDistance(distance);
+            workout.setAverageHR(avgHR);
+            workout.setAverageSpeed(avgSpeed);
+            workout.setCaloriesRate(calRate);
+            workout.setCaloriesBurned(calTotal);
+
+            return workout; //returns completed workout
+        } catch (Exception error) {
+            Log.d(TAG, "Attempting to retrieve Course data: " + error.getMessage());
+        } finally {
+        if (cursor != null) cursor.close();
+        db.close();
+    }
+        return workout; //Returns NULL WORKOUT if the cursor doesn't find the workout, or exception thrown
+    }
+
+    /*
+    Returns all workouts currently saved from the fitness fragment.
+    Returns empty list if there are no workouts
+     */
+    public List<Workout> getWorkouts(){
+        SQLiteDatabase db = this.getReadableDatabase();
+        Cursor cursor = null;
+        Log.d(TAG,"getWorkouts");
+
+        try {
+            cursor = db.query(DbContract.WorkoutEntry.TABLE_NAME, null, null, null, null, null, null);
+
+            if (cursor != null) {
+                cursor.moveToFirst();   //initializes cursor at table start
+                ArrayList<Workout> workoutList = new ArrayList<>();
+                do {
+                    //Stores all columns in the table as a new entry to the arraylist
+                    Workout workout = new Workout();
+
+                    //retrieve data from db
+                    /*
+                    DESERIALIZING:
+                        - Strings stored in DB are taken and converted back to Lists using the Collections
+                        object TypeToken.
+                        - Converted lists are then stored in workout class as normal.
+                        - Uses GSON api.
+                    */
+                    Log.d(TAG,"Deserializing");
+                    Gson gson = new Gson();
+                    String JSONtime = cursor.getString(cursor.getColumnIndex(DbContract.WorkoutEntry.COLUMN_TIME_LIST));
+                    String JSONhr = cursor.getString(cursor.getColumnIndex(DbContract.WorkoutEntry.COLUMN_HR_LIST));
+                    String JSONspeed = cursor.getString(cursor.getColumnIndex(DbContract.WorkoutEntry.COLUMN_SPEED_LIST));
+
+                    Type listType_long = new TypeToken<Collection<Long>>(){}.getType();
+                    Type listType_double = new TypeToken<Collection<Double>>(){}.getType();
+                    List<Long> time = gson.fromJson(JSONtime,listType_long);
+                    List<Double> heartRate = gson.fromJson(JSONhr,listType_double);
+                    List<Double> speed = gson.fromJson(JSONspeed,listType_double);
```

Comment:
@BDeslippe23 , everything looks great. Did you manage to print these values to make sure it retrieves the lists correctly?

Reply:
@DLabbate I just checked it, the list is retrieved properly.